### ISSUE TITLE
Add ability to request a block via its hash 

### ISSUE THREAD

**Issue by evan-forbes** [celestiaorg/celestia-core#1395]
We should add the ability to query a block by its hash via grpc. Currently, we can only easily by its height.

To close this issues, introduce a new grpc method that uses a new index `hash: height`, to first look up the height and then look up the block based on its height.

ref https://github.com/celestiaorg/celestia-app/issues/3421#issuecomment-2094147107

**Comment by cmwaters** [comment]
I don't think this issue belongs in this repo. We discussed having the gRPC server as a service in `celestia-app` or at least extending the existing interface in our `cosmos-sdk` although I would prefer to keep this in celestia-app

**Comment by Eoous** [comment]
I think it's no necessary and difficult to have gRPC service in `celestia-app` or `cosmos-sdk`.

Looks like that we can merge gRPC service from `cometbft` for `celestia-core`: https://github.com/cometbft/cometbft/issues/2693

**Comment by rach-id** [comment]
Does https://github.com/celestiaorg/celestia-core/pull/1513 closes this too?
